addons/common/persistence/jdbc/src/main/java/org/kie/kogito/persistence/jdbc/JDBCProcessInstances.java

```diff
@@ -45,9 +44,30 @@
 
 public class JDBCProcessInstances implements MutableProcessInstances {
 
-    private static final String VERSION = "version";
+    static final String PAYLOAD = "payload";
+    static final String VERSION = "version";
 
-    private static final String PAYLOAD = "payload";
+    enum DatabaseType {
+        POSTGRES("PostgreSQL", "process_instances"),
+        ORACLE("Oracle", "PROCESS_INSTANCES");
+
+        private final String dbIdentifier;
+        private final String tableNamePattern;
+
+        DatabaseType(final String dbIdentifier, final String tableNamePattern) {
+            this.dbIdentifier = dbIdentifier;
+            this.tableNamePattern = tableNamePattern;
+        }
+
+        public static DatabaseType create(final String dbIdentifier) {
+            if ("Oracle".equals(dbIdentifier)) {
+                return ORACLE;
+            } else if ("PostgreSQL".equals(dbIdentifier)) {
+                return POSTGRES;
+            }
+            return null;
```

Comment:
I think it is better to have a default return value in any case, we can decide to have Postgres as default or any other but I would avoid to return null because it makes impossible to even try a different DB and see if it works (with proper logging message)
```suggestion
            log.info("No specific match found for DB identifier '" + dbIdentifier + "', assuming PostgreSQL as default"); 
            return POSTGRES;
```